ospf6d/ospf6_top.c

```diff
@@ -1078,48 +1182,54 @@ static int config_write_ospf6(struct vty *vty)
 	struct listnode *j, *k;
 	struct ospf6_area *oa;
 	struct ospf6_interface *oi;
+	struct ospf6 *ospf6;
+	struct listnode *node, *nnode;
 
 	/* OSPFv3 configuration. */
-	if (ospf6 == NULL)
+	if (om6 == NULL)
 		return CMD_SUCCESS;
 
-	inet_ntop(AF_INET, &ospf6->router_id_static, router_id,
-		  sizeof(router_id));
-	vty_out(vty, "router ospf6\n");
-	if (ospf6->router_id_static != 0)
-		vty_out(vty, " ospf6 router-id %s\n", router_id);
-
-	/* log-adjacency-changes flag print. */
-	if (CHECK_FLAG(ospf6->config_flags, OSPF6_LOG_ADJACENCY_CHANGES)) {
-		if (CHECK_FLAG(ospf6->config_flags, OSPF6_LOG_ADJACENCY_DETAIL))
-			vty_out(vty, " log-adjacency-changes detail\n");
-		else if (!SAVE_OSPF6_LOG_ADJACENCY_CHANGES)
-			vty_out(vty, " log-adjacency-changes\n");
-	} else if (SAVE_OSPF6_LOG_ADJACENCY_CHANGES) {
-		vty_out(vty, " no log-adjacency-changes\n");
-	}
+	for (ALL_LIST_ELEMENTS(om6->ospf6, node, nnode, ospf6)) {
+		inet_ntop(AF_INET, &ospf6->router_id_static, router_id,
+			  sizeof(router_id));
+		vty_out(vty, "router ospf6\n");
+		if (ospf6->router_id_static != 0)
+			vty_out(vty, " ospf6 router-id %s\n", router_id);
```

Comment:
We can remove the `inet_ntop` above and use the format option `%pI4` here.

E.g.: `vty_out(vty, " ospf6 router-id %pI4\n", &ospf6->router_id_static);`

Reply:
Resolved